Rewrite (#8)
Sydney-Informatics-Hub/labarchives-stoichiometry-widget: Readme.md

> **The Readme is to detail the behaviour expected from the Stochiometry widget**

Reviewer: I think you need some introductory information above this. Then you can put in a `# Expected behaviour` or `# Requirements` heading

> 1. When initializing the records of the experiment

Reviewer: simple language:

* "When first entering data into the table"

2. When changing the values of the experiment

> **Initializing the records of the experiment:**

Reviewer: Why not use `##`-style headings?

> On launching the Stochiometry widget, to add to the page, all rows, except for row 1

Reviewer: avoid negation: "only row 1 will be enabled for entry, although subsequent rows will be displayed"

will be disabled for entry.

For row 1 initial entry, the following are the permitted ways - 

|  User Enters  | Automatically Calculated |
| ------------- | ------------- |

> | amt and fw  | moles  |

Reviewer: If we are distinguishing this from "fw and amt" we should probably change "and" to "then"

| fw and amt  | moles |
| moles and fw | amt  |
| fw and moles  | amt  |

The equiv value for row 1 is set to 1 and cannot be edited by the user.

For rows greater than 1 for initial entry, the following are the permitted ways -

|  User Enters  | Automatically Calculated |
| ------------- | ------------- |

> | equiv and fw  | mol and amt  |

Reviewer: "equiv and fw" -> "equiv then fw"

| fw and equiv  | mol and amt |
| fw and amt | mol and equiv  |
| amt and fw  | mol and equiv  |

> Once the first row and the rows needed for experiment are filled, add the Stochiometry to the page.

Reviewer: I don't think this is relevant to our requirements, although we could have a section on usage above? We have the broad requirement (which we discussed in our first meeting, IIRC) that that state should be saved and restored exactly if we click save and edit again, no matter how much has been edited so far.

> **Changing the values of the records of the experiment:**

Reviewer: above you do not say "records of the" in a similar statement

For row 1, the behaviour is as below - 

|  User Changes  | Automatically Calculated |
| ------------- | ------------- |

> | amt  | mol and amt are recalculated for all rows, greater than 1, which have valid values |

Reviewer: "all rows greater than 1" is much clearer than "all rows, greater than 1"

"all non-initial rows" is better than "all rows greater than 1"

Reviewer: mol in the first row must also be recalculated.

| mol  | mol and amt are recalculated for all rows, greater than 1, which have valid values |

For rows greater than 1, the behaviour is as below - 

|  User Changes  | Automatically Calculated |
| ------------- | ------------- |
| amt  | equiv and moles are recalculated for the row for which the amt was changed |
| equiv  | amt and moles are recalculated for the row for which the equiv was changed  |
| fw  | amt and moles are recalculated for the row for which the fw was changed  |

**For volume and density fields**

> For all rows, user has to enter the density in the experiment, the volume will be automatically calculated by the formula -

Reviewer: This makes it sound like the user needs to enter density in all rows. They do not. Density is not helpful for solids.

I can't remember if we also want to be able to calculate mols from volume and density...

> The Readme is to detail the behaviour expected from the Stochiometry widget. Stochiometry widget is to 

Reviewer: ```suggestion
# Stoichiometry Widget for LabArchives eNotebook

Stochiometry widget is to 
```

> The Readme is to detail the data entry process and the behaviour expected from the Stochiometry widget for LabArchives eNotebook. 

Reviewer: I think this should start with a heading, as it did in a previous comment of mine.
